File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        std::istringstream iss(s);
        std::vector<std::string> words;

        // Tokenize the string into words
        while (iss >> s) {
            words.push_back(s);
        }

        // Reconstruct the string with reversed words
        s = "";
        for (auto it = words.rbegin(); it != words.rend(); ++it) {
            s += *it;
            if (it + 1 != words.rend()) {
                s += " ";
            }
        }

        return s;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String ans="";
        for(int i=0;i<s.length();i++){
            StringBuilder str=new StringBuilder();
            while(i<s.length() && (s.charAt(i)!=' ')){
                str.append(s.charAt(i));
                i++;
            }
            if(str.length()!=0){
                ans=" "+str+ans;
            }
        }
        return ans.substring(1);
    }
}